platform-sdk/swirlds-common/src/main/java/com/swirlds/common/config/sources/MappedConfigSource.java

```diff
@@ -0,0 +1,135 @@
+ * mappings for given properties. This allows the same property value defined for one name in the wrapped
+ * {@link ConfigSource} to be accessed by both names.
+ * <p>
+ * For example, suppose you want to rename the "dbUrl" property to "general.databaseUrl". You can use a
+ * {@code MappingConfigSource} to create a mapping between "dbUrl" and "general.databaseUrl", so that you can use both
+ * names with the same value during the transition time.
+ * </p>
+ * <p>
+ * Note that multiple mappings defined for a property is not allowed and will throw an {@link IllegalArgumentException}
+ * at runtime.
+ * </p>
+ * <p>
+ * Note that adding a mapping to a property not defined in the original {@link ConfigSource} will throw an
+ * {@link IllegalArgumentException} at runtime.
+ * </p>
+ * <p>
+ * Note that the ordinal of this {@code ConfigSource} is taken from the original {@link ConfigSource} that was wrapped
+ * by this class.
+ * </p>
+ *
+ * @see ConfigSource
+ * @see ConfigMapping
+ */
+public class MappedConfigSource extends AbstractConfigSource {
+
+    private static final Logger logger = LogManager.getLogger(MappedConfigSource.class);
+
+    private final ConfigSource wrappedSource;
+
+    private final Queue<ConfigMapping> configMappings;
+    private Map<String, String> properties;
+
+    /**
+     * Constructor that takes the wrapped config.
+     *
+     * @param wrappedSource the wrapped config
+     */
+    public MappedConfigSource(@NonNull final ConfigSource wrappedSource) {
+        this.wrappedSource = ArgumentUtils.throwArgNull(wrappedSource, "wrappedSource");
+        this.configMappings = new ConcurrentLinkedQueue<>();
+    }
+
+    /**
+     * Adds the mappedName {@code 'mappedName'<->'originalName'}
+     *
+     * @param mappedName   the mappedName name
+     * @param originalName the original name
+     */
+    public void addMapping(@NonNull final String mappedName, @NonNull final String originalName) {
+        addMapping(new ConfigMapping(mappedName, originalName));
+    }
+
+    /**
+     * Adds the mappedName {@code 'mappedName'<->'originalName'}
+     *
+     * @param configMapping defined mapping
+     */
+    public void addMapping(@NonNull final ConfigMapping configMapping) {
+        configMappings.add(configMapping);
+    }
+
+    @Override
+    @NonNull
+    protected Map<String, String> getInternalProperties() {
+        if (properties == null) {
+            final Map<String, String> internalProperties = wrappedSource.getProperties();
+            final Map<String, String> mappedProperties = new HashMap<>();
+            final Set<String> originals = new HashSet<>();
+
+            configMappings.forEach(configMapping -> {
+                if (internalProperties.containsKey(configMapping.mappedName())) {
+                    throw new IllegalArgumentException(
+                            "Property '" + configMapping.mappedName() + "' already found in original config source");
```

Comment:
nitpick.
This might read better if the message is a constant that's formatted.
Example:
```java
private static final String MAPPING_FOR_EXISTING_KEY = "Value '%s' already present in original configuration source.";
...
throw new IllegalArgumentException(MAPPING_FOR_EXISTING_KEY.formatted(configMapping.mappedName()));
```